src/camera_utils.py

```diff
@@ -46,7 +54,21 @@ def compute_image_footprint_on_surface(
     Returns:
         [footprint_x, footprint_y] in meters as a 2-element array.
     """
-    raise NotImplementedError()
+    Z = distance_from_surface
+    #project top left corner of the image (0,0)
+    x0 = (0 - camera.cx) * Z / camera.fx
+    y0 = (0 - camera.cy) * Z / camera.fy
+
+    #reproject (width,height)
+    x1 = (camera.image_size_x_px - camera.cx) * Z / camera.fx
+    y1 = (camera.image_size_y_px - camera.cy) * Z / camera.fy
```

Comment:
Nice idea to use reprojection equations, but I will recommend you to implement the function in the bonus part, and just reuse the function here.